README.md

```diff
@@ -435,6 +355,347 @@ One App can be started via docker or if built from source by running `node lib/s
+* [Frank Lloyd Root's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
+
+##### `createSsrFetch`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+RootModule.appConfig = {
+  createSsrFetch: ({
+    req, // Express req
+    res, // Express res
+  }) => (fetch) => (fetchUrl, fetchOpts) => Promise,
+};
+``` 
+
+`createSsrFetch` allows for customizing the fetch client used in `one-app` to perform server-side requests. 
+
+For example, you may wish to forward cookies or headers from the initial page load request to all the requisite SSR API requests.
+
+**📘 More Information**
+* Example: [Frank Lloyd Root's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
+* Example: [An SSR Fetch Client](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/createFrankLikeFetch.js)
+* Using SSR Fetch Client with [`loadModuleData`](#loadModuleData)
+
+##### `validateStateConfig`
+**Module Type**
+* 🚫 Root Module 
+* ✅ Child Module 
+
+**Shape**
+```js
+Module.appConfig = {
+  validateStateConfig: {
+    server: {
+      [settingName]: {
+        validate(settingValue) {
+          // Throw an error or return undefined
+        },
+      },
+    },
+    client: {
+      [settingName]: {
+        validate(settingValue) {
+          // Throw an error or return undefined
+        },
+      },
+    },
+  },
+};
+```
+
+The `validateStateConfig` allows a Child Module to validate settings passed from `provideStateConfig`. Each `settingName` object accepts a `validate(settingValue)` method. The `validate` function may throw an `Error` or return `undefined` depending on validity of the value supplied to the Module on load.
+
+If an `Error` is thrown, the server will prevent the child module from being loaded.
+
+**📘 More Information**
+* [`provideStateConfig`](#providestateconfig)
+* Example: [Picky Frank's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/picky-frank/0.0.0/src/components/PickyFrank.jsx)
+
+##### `requiredSafeRequestRestrictedAttributes`
+**Module Type**
+* 🚫 Root Module 
+* ✅ Child Module 
+
+**Shape**
+```js
+Module.appConfig = {
+  requiredSafeRequestRestrictedAttributes: {
+    headers: [String],
+    cookies: [String],
+  },
+};
+```
+
+The `requiredSafeRequestRestrictedAttributes` allows a Child Module to validate settings passed from `extendSafeRequestRestrictedAttributes`. Each whitelisted header in `headers` array and cookie in `cookies` array will be checked against the Root Module's `extendSafeRequestRestrictedAttributes` on the loading of the Child Module. If this does not match entries previously made in Root Module's `extendSafeRequestRestrictedAttributes`, the Child Module will fail to load.
```

Comment:
We should add a bit on why failing to load is important (rather than letting it fail at runtime). The same bit should be added to all the other "type checking" APIs.

Reply:
also prevents app startup

Reply:
Fixed in 7bc9f58